jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -1163,4 +1163,180 @@ else if (l.size() == 1)
             s -> s.map(Objects::toString).orElse("")
         );
     }
+
+    /**
+     * Realize of linear regression functions.
+     */
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrCountLong()</code> function.
+     */
+    public static  Collector<Tuple2<Double,Double>, ?, Optional<Long>> regrCountLong() {
+        return regrCountLong(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrCountLong()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Long>> regrCountLong(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(toList(), l -> {
+            if (l.isEmpty())
+                return Optional.empty();
+            long result = 0;
+            for (T t : l)
+                result += t==null ? 0:1;
+            return Optional.of(result);
+        });
+    }
```

Comment:
About these ones:

1. Why `Long` when the input and calculation is `Double` based?
2. What's the point of passing `functionX` and `functionY` when they're not used?
3. In SQL, `REGR_COUNT(x, y)` makes some sense because of the implied `NULL` semantics, though even there, it can be trivially emulated using `COUNT(x + y)`, so it is of limited use. In Java, the `null` semantics can be achieved equally trivially using `Collectors.filtering(v -> v != null, Collectors.counting())`, so I'm not convinced this is really useful
4. Given the above emulation, we shouldn't operate on a pre-collected list and then iterate over it again.

Reply:
Overall, I have some doubts about the utility of some of these, but it wasn't well specified, and people might expect SQL aggregate functions in this library without much change in semantics - that's fine. So, let's ignore bullet 2 for now.

Reply:
Also, returning `Long` is probably the right thing to do here, thinking about it again. It's consistent with `Agg.count()`